scipy/stats/_multivariate.py

```diff
@@ -155,9 +155,12 @@ def __init__(self, M, cond=None, rcond=None, lower=True,
         # Compute the symmetric eigendecomposition.
         # Note that eigh takes care of array conversion, chkfinite,
         # and assertion that the matrix is square.
+        M = np.asarray(M)
         s, u = scipy.linalg.eigh(M, lower=lower, check_finite=check_finite)
 
         eps = _eigvalsh_to_eps(s, cond, rcond)
+        if not np.allclose(M, M.T, atol=1e-10):
+            raise ValueError('the input matrix must be symmetric')
```

Comment:
Do we need `M.T` anywhere else in this method/function? Like, is `M.T` done only for this check?

Reply:
Yes, `M.T` only needed for this check. I converted M into a numpy array because I thought this was an efficient way to check symmetric matrix.

Reply:
Well, in other checks of this method/function, I noticed that such information is used which is anyways needed even if those checks are removed. So, I am not sure if its a good idea to use M.T and then doing `np.allclose(M, M.T)` (extra O(n^2) which is non-trivial I think) for this error message. If user passes a non-symmetric matrix then it would be a case of "Garbage In Garbage Out". Though this is my personal opinion.

Reply:
I referred to [`sklearn.utils.validation.check_symmetric](https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/utils/validation.py#L1092)

Reply:
If I look it that way, I think you are right. M matrix is supposed to be symmetric when passed by the user. I will close this PR. What do you suggest we do with issue #15508?

Reply:
User's matrix in #15508 is a positive semidefinite matrix (all eigen values are non-negative) but not a positive definite matrix (one eigen value is 0). Multivariate normal's PDF exists only if M is positive definite (according to https://en.wikipedia.org/wiki/Multivariate_normal_distribution). So, IMO, the condition at line 164 should be changed to `if np.min(s) <= 0` and the error message to, `'the input matrix must be positive definite'`.

Though note that its confusing. In the same Wikipedia page, M is required to be just positive semidefinite.

Reply:
The PDF is _degenerate_ in the full space if the matrix has some 0 eigenvalues. But then it's just a multivariate normal in a lower-dimensional subspace of the full space. For the PDF formula to be finite, positive-definiteness is required. But for the purpose of random number generation, we don't care about the PDF being nice and finite (and there have been requests to allow such degenerate cases); we use a method that samples in the lower-dimensional subspace just fine.